### PR TITLE
BUGFIX: Wrong usage of delete operator in Settings.load

### PR DESCRIPTION
**Pull request by catloversg**
I found this bug while implementing #1789.

Test code:
```js
/** @param {NS} ns */
export async function main(ns) {
  ns.tail();
  const object = {
    prop1: 1,
    prop2: 2,
  };
  delete object.prop1, object.prop2; // Wrong
  // delete object.prop1; delete object.prop2; // Correct
  ns.print(object);
}
```

### CODE REVIEW

**Comment by d0sboots** [comment]
Slightly surprised lint doesn't complain about this, but I guess it's considered covered by "don't use delete."

File: src/Settings/Settings.ts [hunk 1]
```diff
@@ -164,7 +164,10 @@ export const Settings = {
     save.styles && Object.assign(Settings.styles, save.styles);
     save.overview && Object.assign(Settings.overview, save.overview);
     save.EditorTheme && Object.assign(Settings.EditorTheme, save.EditorTheme);
-    delete save.theme, save.styles, save.overview, save.EditorTheme;
+    delete save.theme;
+    delete save.styles;
+    delete save.overview;
+    delete save.EditorTheme;
     Object.assign(Settings, save);
```
Comment (d0sboots) [inline src/Settings/Settings.ts]: If the `assign` is written this way, the undesirable save properties will be re-overwritten with the ones from Settings, eliminating the need to delete anything (and thus avoiding a future lint issue).
```suggestion
    Object.assign(Settings, save, {Settings.theme, Settings.styles, Settings.overview, Settings.EditorTheme});
```

Edit: On the whole, the use of assign here is a bit awkward, but possibly necessary if things are keeping references to the settings object. And it's hard to know (without auditing the whole codebase) if that's hapenning.

Comment (d0sboots) [reply src/Settings/Settings.ts]: I see you spelled out all the keys, did the shorthand version not work for some reason?

Comment (catloversg) [reply src/Settings/Settings.ts]: The shorthand version is a syntax error.
Test code:
```js
/** @param {NS} ns */
export async function main(ns) {
  const Settings = {
    theme: {
      color: "#fff"
    }
  };
  const save = {
    theme: {
      color: "#000"
    }
  };
  save.theme && Object.assign(Settings.theme, save.theme);
  // Object.assign(Settings, save, { Settings.theme }); // Syntax error
  Object.assign(Settings, save, {
    theme: Settings.theme,
  });
}
```

**Pull request merged**